babel-node error when passing arguments

<!---
BEFORE YOU SUBMIT please read the following:

Please search open/closed issues before submitting since someone might have asked the same thing before!

If you have a support request or question please submit them to one of this resources:

* Discussion Forum: https://discuss.babeljs.io/
* StackOverflow: http://stackoverflow.com/questions/tagged/babeljs using the tag `babeljs`
* Slack community: https://slack.babeljs.io/
* Also have a look at the Readme for more information on how to get support:
  https://github.com/babel/babel/blob/master/README.md

Issues on GitHub are only related to problems of babel itself and we cannot answer 
support questions here.
-->

<!--- Provide a general summary of the issue in the title above -->

### Input Code
<!--- If you're describing a bug, please let us know which sample code reproduces your problem -->
<!--- If you have link to our REPL or a standalone repo please link that! -->
When trying to run index.js with arguments, babel-node will error out.

```bash
babel-node index.js test
```

### Babel Configuration (.babelrc, package.json, cli command)
<!--- If describing a bug, tell us what your babel configuration looks like -->

Config is not invoked as babel-node fails to identify the correct path to index.js
```js
{
  "your": { "config": "here" }
}
```

### Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
babel-node runs index.js with "test" in the argument list


### Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

The following Error is shown:
```
module.js:472
    throw err;
    ^

Error: Cannot find module '/Users/jack.huang/github/u9520107/ringcentral-js-integration-commons/test'
    at Function.Module._reso lveFilename (module.js:470:15)
    at Function.Module._load (module.js:418:25)
    at Function.Module.runMain (module.js:605:10)
    at /Users/jack.huang/.nvm/versions/node/v7.2.0/lib/node_modules/babel-cli/lib/_babel-node.js:155:24
    at Object.<anonymous> (/Users/jack.huang/.nvm/versions/node/v7.2.0/lib/node_modules/babel-cli/lib/_babel-node.js:156:7)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
```


### Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
This affects our project build script as we write gulpfiles with es6. We currently do things like the following, and would run into this bug.

```bash
./node_modules/babel-cli/bin/babel-node.js ./node_modules/gulp/bin/gulp.js test
```



### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

| software         | version
| ---------------- | -------
| Babel            |  6.22.1
| node             | 7.2.0
| npm              |  3.10.9  (also used yarn 0.19.1)
| Operating System |  maxos (also tested on linux, but not sure which distro)

we are seeing the same problem where previously this command:

```
babel-node ~/src/main.js --dir /bla/static/js --files ./base/preconditions.js ./base/serialization.js ./services/print/dummy_print_service.js ./services/print/print_proto.js ./services/print/print_service.js --entry-points services/print/dummy_print_service.js
```

would work, but now errors with:
```
Error: Cannot find module '/Users/joscha/dev/canva/cfe/services/print/dummy_print_service.js'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Function.Module.runMain (module.js:590:10)
    at /Users/joscha/dev/canva/tools/gendeps/node_modules/babel-cli/lib/_babel-node.js:155:24
    at Object.<anonymous> (/Users/joscha/dev/canva/tools/gendeps/node_modules/babel-cli/lib/_babel-node.js:156:7)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
```

--- a/packages/babel-cli/src/_babel-node.js
+++ b/packages/babel-cli/src/_babel-node.js
@@ -103,7 +103,7 @@ if (program.eval || program.print) {
 
     let i = 0;
     let ignoreNext = false;
-    args.forEach(function (arg, i2) {
+    args.some(function (arg, i2) {
       if (ignoreNext) {
         ignoreNext = false;
         return;
@@ -116,7 +116,7 @@ if (program.eval || program.print) {
         }
       } else {
         i = i2;
-        return false;
+        return true;
       }
     });
     args = args.slice(i);

